test/JDBC/src/test/java/com/sqlsamples/TestQueryFile.java

```diff
@@ -396,28 +406,54 @@ public void TestQueryBatch(String inputFileName) throws SQLException, ClassNotFo
         BufferedWriter bw = new BufferedWriter(fw);
         batch_run.batch_run_sql(connection_bbl, bw, testFilePath, logger);
         bw.close();
-        
+        if(sla == 0){
+            sla = defaultSLA*1000000L;
+        }
+        int n = exec_times.size();
+        int index = n - num_lines;
+        long total_time = 0;
+        for(int i = index; i < n; i++){
+            total_time += exec_times.get(i);
+        }
         File expectedFile = new File(generatedFilesDirectoryPath + outputFileName + ".out");
         File sqlExpectedFile = new File(sqlServerGeneratedFilesDirectoryPath + outputFileName + ".out");
 
-        if (expectedFile.exists()) {
+        if(total_time <= sla){
+            timeout = false;
+        }
+        else{
+            timeout = true;
+        }
+        if (!timeout && expectedFile.exists()) {
             // get the diff
             result = compareOutFiles(outputFile, expectedFile);
-        } else if (sqlExpectedFile.exists()) {
+        } else if (!timeout && sqlExpectedFile.exists()) {
             // get the diff
             result = compareOutFiles(outputFile, sqlExpectedFile);
         } else {
             result = false;
         }
 
+        ArrayList<Long> tempSla = new ArrayList<>();
+        tempSla.add(total_time/1000000L);
+        tempSla.add(sla/1000000L);
         summaryMap.add(new AbstractMap.SimpleEntry<>(inputFileName, result)); //add test name and result to map
+        slaMap.add(new AbstractMap.SimpleEntry<>(inputFileName, tempSla)); //add execution time and SLA
+        sla = 0L;
 
         try {
             Assertions.assertTrue(result);
         } catch (AssertionError e) {
-            Throwable throwable = new Throwable(inputFileName + " FAILED! Output diff can be found in '" + diffFile.getAbsolutePath() + "'");
-            throwable.setStackTrace(new StackTraceElement[0]);
-            throw throwable;
+            if(timeout){
```

Comment:
There should be a way to distinguish if a test failed solely due to timeout or if it failed due to both a diff and a timeout. From this logic, it seems that if a test times out, we just say Execution timed out but we do not see if there was potentially a diff as well.